### PR TITLE
Added extensiondev command and prepare_python task

### PR DESCRIPTION
**Pull request by MaciejGrynda**

### CODE REVIEW

File: dtcli/dev.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            args.extend(['-f', additional_path])
+
+        args.append(setup_path)
+        result = subprocess.run(args, capture_output=True)
+
+        if result.returncode != 0:
+            print("Error building python extension: {}".format(result.stderr.decode('utf-8')), file=sys.stderr)
+            return result.returncode
+
+        lib_folder = target_path + '/lib'
```
Comment (wbachnik) [inline dtcli/dev.py:20]: Please use `os.path.join()` to combine path elements.

File: dtcli/dev.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import os
+
+def pack_python_extension(setup_path, target_path, additional_path):
+    with tempfile.TemporaryDirectory() as tmp:
+        args = [sys.executable, '-m', 'pip', 'wheel', '-w', tmp]
+        if additional_path is not None:
+            args.extend(['-f', additional_path])
+
+        args.append(setup_path)
+        result = subprocess.run(args, capture_output=True)
```
Comment (wbachnik) [inline dtcli/dev.py:14]: Can't we avoid using subprocess in favor of using the `pip` module directly? Wouldn't that be simpler, esp. wrt error handling?

File: dtcli/scripts/dt.py [hunk 3]
```diff
@@ -170,5 +170,32 @@ def build(**kwargs):
     )
 
 
+@extension_dev.command(
+    help="comand packs python package as a datasource. It uses pip to download all dependencies and create whl files"
```
Comment (wbachnik) [inline dtcli/scripts/dt.py:174]: I'd describe that simply with "builds Python extension".

**Review by wbachnik — CHANGES_REQUESTED** [review]

File: dtcli/dev.py (hunk 2 above)
Comment (MaciejGrynda) [reply dtcli/dev.py:14]: Pip actively discourages to to be used directly:
https://stackoverflow.com/questions/47059526/should-i-use-pip-main-or-subprocess-call-to-invoke-pip-commands

File: dtcli/scripts/dt.py (hunk 3 above)
Comment (MaciejGrynda) [reply dtcli/scripts/dt.py:174]: I was thinking about it, but we have "dt ext build" command, which creates whole extension pack, and I decided to not use this name

File: dtcli/dev.py (hunk 2 above)
Comment (wbachnik) [reply dtcli/dev.py:14]: Hmm, it seems it's actually not recommended to use `pip` programmatically, any experience with that?

Comment (MaciejGrynda) [reply dtcli/dev.py:14]: I didn't try it, I've only done a research. It should be possible, but pip can change it's main() interface.
In EXT 1.0 plugins SDK we have also used subprocess to create plugin builds

Comment (wbachnik) [reply dtcli/dev.py:14]: OK, I'm convinced :-)

**Review by vduseev — DISMISSED** [review]

**Review by wbachnik — APPROVED** [review]
Just please remove "comand" from the command description to keep it consistent with other descriptions.

**Pull request merged**